### PR TITLE
S2218 publisherid

### PR DESCRIPTION
**Pull request by hjeeves**
caom2-search-server v 2.5.

Preview link and one-click download use Publisher ID. Publisher ID is available as a column in the results page. New registry capabilities (supporting data collections as resources) supported. 

wwebui branch s2218 contains AdvancedSearch & MaqSearch with build files using this library version.

### CODE REVIEW

File: caom2-search-server/src/main/java/ca/nrc/cadc/search/PackageServlet.java [hunk 1]
```diff
@@ -119,16 +120,34 @@ protected void doGet(final HttpServletRequest request, final HttpServletResponse
...
+        String[] idValues = request.getParameterValues("ID");
+        if (idValues.length > 1) {
+            throw new UnsupportedOperationException("Multiple IDs in package lookup.");
+        }
+
+        String IDValue = idValues[0];
+        if (IDValue.length() > 0 ) {
+            // split the ID parameter on '?' to pull off the query string.
+            // use the first part (resourceid) in the getServiceURL
+            String[] uri_parts = IDValue.split("\\?");
```
Comment (pdowler) [inline caom2-search-server/src/main/java/ca/nrc/cadc/search/PackageServlet.java]: There is a class called PublisherID in the caom2 library that will parse the URI and extract the resourceID for you...

File: caom2-search-server/src/main/java/ca/nrc/cadc/search/SearchPreviewServlet.java [hunk 2]
```diff
@@ -71,13 +76,39 @@ public SearchPreviewServlet(final PreviewRequestHandler previewRequestHandler) {
...
+        final String uriStr = req.getParameter("id");
+        URL serviceURL = null;
+        profiler.checkpoint(String.format("%s doGet() start", uriStr));
+
+        // PublisherID format expected: ivo://<resourceID>?<query string>
+        if (uriStr.length() > 0) {
+            try {
+                // split the ID parameter on '?' to pull off the query string.
+                // use the first part (resourceid) in the getServiceURL.
+                String[] uri_parts = uriStr.split("\\?");
```
Comment (pdowler) [inline caom2-search-server/src/main/java/ca/nrc/cadc/search/SearchPreviewServlet.java]: as above, you could use PublisherID

File: caom2-search-server/src/main/java/ca/nrc/cadc/search/SearchPreviewServlet.java [hunk 3]
```diff
@@ -71,13 +76,39 @@ public SearchPreviewServlet(final PreviewRequestHandler previewRequestHandler) {
...
-            getServiceID(
-                CAOM2LINK_SERVICE_URI_PROPERTY_KEY,
-                DEFAULT_CAOM2LINK_SERVICE_URI),
-            Standards.DATALINK_LINKS_10, AuthMethod.COOKIE);
-        this.previewRequestHandler = new PreviewRequestHandler(dataServiceURL, new JobURLCreator() {
+        final String uriStr = req.getParameter("id");
+        URL serviceURL = null;
+        profiler.checkpoint(String.format("%s doGet() start", uriStr));
+
+        // PublisherID format expected: ivo://<resourceID>?<query string>
```
Comment (pdowler) [inline caom2-search-server/src/main/java/ca/nrc/cadc/search/SearchPreviewServlet.java]: the scheme is part of the resourceID so this comment is wrong: <resourceID>?<query string>

**Comment by at88mph** [comment]
Code pull #90 incorporates this one, and adds code review cleanup.

**Pull request closed without merging**